src/EventListener/Doctrine/RemoveListener.php

```diff
@@ -54,5 +54,6 @@ public function postFlush(): void
                 $this->handler->remove($object, $field);
             }
         }
+        $this->entities = [];
```

Comment:
What's the purpose of this line?

Reply:
It resets the internal state. It removes the references to the entities and frees memory, and if the listener runs multiple times the handler will not be called again for the same entities.

Reply:
Can you provide an example of a case where the listener runs multiple times?

Reply:
Of course, for example if the entity manager flushed multiple times, e.g. in a batch process or at context/transaction boundaries (and apart from that I think making services as stateless as possible is a best practice).

```php
$photos = $entityManager->getRepository(FeaturedPhoto::class)->findAll();

$entityManager->remove(array_shift($photos));
$entityManager->flush();

$entityManager->remove(array_shift($photos));
$entityManager->flush();
```

![image](https://user-images.githubusercontent.com/162986/176004724-82f007c7-ae41-43e1-ba83-8a0fd4404db1.png)